force-app/main/default/classes/AFFL_MultiRecordTypeMapper.cls

```diff
@@ -330,32 +313,30 @@ public virtual with sharing class AFFL_MultiRecordTypeMapper {
 
     /*******************************************************************************************************
      * @description Whether or not the affiliation requires a program enrollment to be auto-created.
-     * @param affl The affiliation to check for program enrollment creations
+     * @param affiliation The affiliation to check for program enrollment creations
      * @return Whether or not the affiliation requires a program enrollment to be auto-created.
      ********************************************************************************************************/
-    public virtual Boolean needsProgramEnrollment(Affiliation__c affl) {
-        if (affl.Affiliation_Type__c == null) {
+    public virtual Boolean needsProgramEnrollment(Affiliation__c affiliation) {
+        if (affiliation?.Affiliation_Type__c == null) {
             return false;
         }
 
-        if (accTypeToEnrollCreate.get(affl.Affiliation_Type__c) == null) {
-            return false;
-        }
-
-        Boolean autoEnrollment = accTypeToEnrollCreate.get(affl.Affiliation_Type__c);
+        Affl_Mappings__c affiliationMapping = primaryAffiliationMappingsByAccountRecordTypeName.get(
+            affiliation.Affiliation_Type__c
+        );
 
-        if (autoEnrollment == false) {
+        if (affiliationMapping?.Auto_Program_Enrollment__c != true) {
```

Comment:
Note that we're checking to see if the affiliation mapping is null and if its auto enrollment field is not true